ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -73,9 +73,21 @@ private void generateCommands() throws IOException, TemplateException {
         }
 
         if (ConverterUtil.hasContent(spec.getTypes())) {
+            // Generate EnumConstant Types, which Element types can be dependant on
+            // TODO Determine if dependency matters at code generation time... probably not.
             for (final Ds3Type typeEntry : spec.getTypes().values()) {
-                generateType(typeEntry);
-                generateTypeMatcher(typeEntry);
+                if (typeEntry.getElements().isEmpty()) {
```

Comment:
Good point - I did notice of few of those instances while combing through the contract yesterday.  will fix.